gst.so/src/gst_worker.cpp

```diff
@@ -456,49 +440,66 @@ bool GSTWorker::do_gst_stress_test(int *error, std::string *err_description) {
         //Start the timer
         start_time = gpu_blas->get_time_us();
 
-        // run GEMM & wait for completion
+        // run GEMM operation
         gpu_blas->run_blass_gemm(gst_ops_type);
 
-        /* Set callback to be called upon completion of blas gemm operations */
-        gpu_blas->set_callback(blas_callback, (void *)this);
+        // sgemm, dgemm and hgemm operations
+        if(!gst_ops_type.empty()) {
 
-        std::unique_lock<std::mutex> lk(mutex);
-        cv.wait(lk);
+          /* Set callback to be called upon completion of blas gemm operations */
+          gpu_blas->set_callback(blas_callback, (void *)this);
 
-        if(!blas_status) {
-          msg = "[" + action_name + "] " + MODULE_NAME + " " +
-            std::to_string(gpu_id) + " " + " BLAS gemm operations failed !!! ";
-          rvs::lp::Log(msg, rvs::logtrace);
+          std::unique_lock<std::mutex> lk(mutex);
+          cv.wait(lk);
+
+          //End the timer
+          end_time = gpu_blas->get_time_us();
+
+          if(!blas_status) {
+            msg = "[" + action_name + "] " + MODULE_NAME + " " +
+              std::to_string(gpu_id) + " " + " BLAS gemm operations failed !!! ";
+            rvs::lp::Log(msg, rvs::logtrace);
+          }
         }
+        else {
+          // data type (fp8 & bp16) based gemm operations
 
-        //End the timer
-        end_time = gpu_blas->get_time_us();
+          // Wait for GEMM operation to complete
+          gpu_blas->is_gemm_op_complete();
```

Comment:
same as aboce, check in a loop to ensure completion as this is a non blocking call

Reply:
is_gemm_op_complete() call is blocking now as we use hipStreamSynchronize() instead of hipStreamQuery(). So, no busy loop required.